### PR TITLE
KAFKA-3698: Update the message format section.

### PR DESCRIPTION
**Pull request by becketqin**

### CODE REVIEW

**Comment by ijuma** [comment]
Thanks @becketqin, this seems good to me. @junrao, can you please take a look as well?

File: docs/implementation.html [hunk 1]
```diff
@@ -183,10 +187,15 @@
 <pre>
 On-disk format of a message
 
-message length : 4 bytes (value: 1+4+n)
-"magic" value  : 1 byte
+message length : 4 bytes (value: 4+1+1+8+4+K+4+V)
```
Comment (junrao) [inline docs/implementation.html]: We probably should include the offset before message length.

Comment (dpkp) [inline docs/implementation.html]: perhaps worth reiterating here that the message length only is +8 if magic value is > 0 .

Comment (becketqin) [inline docs/implementation.html]: @junrao I added the offset field, but did not mention the relative offset by assuming this is only talking about the shallow message. Is that OK?

**Comment by ijuma** [comment]
@junrao, are you happy with the latest version?

**Comment by junrao** [comment]
Thanks for the patch. LGTM

**Pull request closed without merging**